### ISSUE TITLE
I have a problem

### ISSUE THREAD

**Issue by jorgebg2016** [aaugustin/websockets#829]
I need to send datas to clients calling a function outside loop handler. Somebody can help me with that or show me an
 example about it?

**Comment by Girish400** [comment]
same problem, Is it even possible?

**Comment by aaugustin** [comment]
I'm not quite sure about what you're asking for. Could you try to rephrase your question with more details? Perhaps show how you tried to implement this and where you're blocked?

**Comment by aaugustin** [comment]
> I' trying to send WebSocket messages outside the handler, Is it possible?

Yes, it's possible.

Your question is weird because you're showing a client and there's no concept of a "handler" for a client. Only the server needs "connection handlers".

Solution #1 

```python
websocket = await websockets.connect(uri)
# here you aren't inside anything, just to whatever you want with websocket
```

Solution #2

```python
stateful_structure = something_from_where_you_will_use_the_websocket_connection()

async with websockets.connect(uri) as websocket:
      stateful_structure.bind(websocket)
      await websocket.wait_closed()
```

The registration example in the docs is a variant of #2 for the server side.

**Comment by lakegirish** [comment]
**Solution A:** As per _Solution A_ my event function would something like this, correct? if yes, How can I  connect only once, so I can receive live data from trading API and whenever data is generated through the event trigger (example: _ib.execDetailsEvent += execDetails_), my event function  (_execDetails_) is called

```
async def newOrder(trade):
    websocket = await websockets.connect(uri)
    await websocket.send("newOrder")


async def execDetails(trade, fill):
    websocket = await websockets.connect(uri)
    await websocket .send("execDetails")


async def orderStatusChange(trade):
    websocket = await websockets.connect(uri)
    await websocket.send("orderStatusChange")
```

Solution B: For binding are you referring [this](https://websockets.readthedocs.io/en/stable/intro.html#synchronization-example) example where you add(bind) user to WebSocket particularly below two lines 
```
 # register(websocket) sends user_event() to websocket
    await register(websocket)
```

_I may sound a little stupid but if time permits can you explain a little more with code. I have a WebSocket server that is connected to the webpage to take buy/sell orders. I want WebSocket Client to collect data from trading API and send it to WebSocket server. Communication between, webpage, Websocket Server, and Websocket Client should be established once to send and receive messages all day. The python library I'm using ib_insync is based and asyncio and so I stumbled upon asyncio WebSockets. Thanks a Ton!!_

**Comment by aaugustin** [comment]
Your version of solution A opens a new connection every time you need to send a message. This isn't how you use WebSocket APIs (in general).

Very basic version: when your program start, do `global websocket; websocket = await websockets.connect(uri)` to store it in a global variable. Then you have the WebSocket connection available everywhere in your program.

This is very basic and won't do everything you want (e.g. what happens when the connection drops?) but it's a good starting point if you have no idea were to start.
